tests/cluster/cluster_connection_pool_test.py

```diff
@@ -126,18 +126,21 @@ async def test_get_connection_by_key():
 
     # Patch the call that is made inside the method to allow control of the
     # returned connection object
-    with patch.object(ClusterConnectionPool, 'get_connection_by_slot',
-                      autospec=True) as pool_mock:
-        def side_effect(*_args, **_kwargs):
-            return SampleConnection(port=1337)
 
-        pool_mock.side_effect = side_effect
+    if tuple(sys.version_info)[:2] <= (3, 7):
```

Comment:
Sorry, I seem to be missing why this behaviour would be different in py3.7? It looks like `get_connection_by_slot()`  is always a coroutine now.